bevm/bevmclient/cmd_imp.go

```diff
@@ -0,0 +1,274 @@
+	if err != nil {
+		return xerrors.Errorf("reading contract Bytecode: %v", err)
+	}
+
+	contract, err := bevm.NewEvmContract("newContract", string(abiData), string(binData))
+	if err != nil {
+		return xerrors.Errorf("creating new BEvm contract: %v", err)
+	}
+
+	userArgs := ctx.Args()[2:]
+	args, err := decodeEvmArgs(userArgs, contract.Abi.Constructor.Inputs)
+	if err != nil {
+		return xerrors.Errorf("decoding contract constructor arguments: %v", err)
+	}
+
+	// Perform command
+
+	contractInstance, err := opt.bevmClient.Deploy(
+		gasLimit, big.NewInt(int64(gasPrice)), amount, opt.account, contract, args...)
+	if err != nil {
+		return xerrors.Errorf("deploying new BEvm contract: %v", err)
+	}
+
+	err = writeAccountFile(opt.account, opt.accountName, false)
+	if err != nil {
+		return xerrors.Errorf("writing account file: %v", err)
+	}
+
+	err = writeContractFile(contractInstance, abiFilepath, contractName, true)
+	if err != nil {
+		return xerrors.Errorf("writing contract file: %v", err)
+	}
+
+	_, err = fmt.Fprintf(ctx.App.Writer, "%s deployed\n", contractInstance)
+	if err != nil {
+		return xerrors.Errorf("writing report msg: %v", err)
+	}
+
+	return nil
+}
+
+func executeTransaction(ctx *cli.Context) error {
+	// Retrieve options and arguments
+
+	opt, err := handleCommonOptions(ctx)
+	if err != nil {
+		return xerrors.Errorf("handling options: %v", err)
+	}
+
+	gasLimit := ctx.Uint64("gas-limit")
+	gasPrice := ctx.Uint64("gas-price")
+	amount := ctx.Uint64("amount")
+	contractName := ctx.String("contract-name")
+
+	contractInstance, err := readContractFile(contractName)
+	if err != nil {
+		return xerrors.Errorf("reading contract file: %v", err)
+	}
+
+	if ctx.NArg() == 0 {
+		return xerrors.New("missing transaction name argument")
+	}
+
+	method := ctx.Args().First()
+	methodAbi, ok := contractInstance.Parent.Abi.Methods[method]
+	if !ok {
+		return xerrors.Errorf("transaction \"%s\" does not exist for this contract", method)
+	}
+
+	userArgs := ctx.Args().Tail()
+	args, err := decodeEvmArgs(userArgs, methodAbi.Inputs)
+	if err != nil {
+		return xerrors.Errorf("decoding contract transaction arguments: %v", err)
+	}
+
+	// Perform command
+
+	err = opt.bevmClient.Transaction(
+		gasLimit, big.NewInt(int64(gasPrice)), amount, opt.account,
+		contractInstance, method, args...)
+	if err != nil {
+		return xerrors.Errorf("executing contract transaction: %v", err)
+	}
+
+	err = writeAccountFile(opt.account, opt.accountName, false)
+	if err != nil {
+		return xerrors.Errorf("writing account file: %v", err)
+	}
+
+	_, err = fmt.Fprintf(ctx.App.Writer, "transaction executed\n")
+	if err != nil {
+		return xerrors.Errorf("writing report msg: %v", err)
+	}
+
+	return nil
+}
+
+func executeCall(ctx *cli.Context) error {
+	// Retrieve options and arguments
+
+	opt, err := handleCommonOptions(ctx)
+	if err != nil {
+		return xerrors.Errorf("handling options: %v", err)
+	}
+
+	contractName := ctx.String("contract-name")
```

Comment:
Actually no, if no contract name is provided, it will default to "contract".
I tried to apply the principle that options are optional :) (with a default value), and arguments are mandatory.